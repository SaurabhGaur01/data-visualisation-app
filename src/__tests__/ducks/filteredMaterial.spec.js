import {
    setFilteredMaterial,
    clearFilteredMaterial,
    filteredMaterialReducer,
} from '../../ducks/filteredMaterial';

describe('filteredMaterial Duck Tests', () => {
    const mockObject = "mockValue";

    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setFilteredMaterial should match the snapshot' , () => {
            expect(setFilteredMaterial(mockObject)).toMatchSnapshot();
        });
        it('clearFilteredMaterial should match the snapshot' , () => {
            expect(clearFilteredMaterial(mockObject)).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(filteredMaterialReducer()).toEqual([]);
        });
        it('should return the passed state if called with no action' , () => {
            expect(filteredMaterialReducer(mockObject)).toEqual(mockObject);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(filteredMaterialReducer(mockObject, unknownAction)).toEqual(mockObject);
        });
        it('should return the value which is passed' , () => {
            expect(filteredMaterialReducer('', setFilteredMaterial(mockObject))).toEqual(mockObject);
        });
        it('should reset the value to []' , () => {
            expect(filteredMaterialReducer('', clearFilteredMaterial())).toEqual([]);
        });
    });
});