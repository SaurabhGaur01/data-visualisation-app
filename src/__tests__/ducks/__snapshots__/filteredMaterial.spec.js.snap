// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`filteredMaterial Duck Tests Action Generator clearFilteredMaterial should match the snapshot 1`] = `
Object {
  "type": "modules/data-visualisation-app/CLEAR_FILTERED_MATERIAL",
}
`;

exports[`filteredMaterial Duck Tests Action Generator setFilteredMaterial should match the snapshot 1`] = `
Object {
  "data": "mockValue",
  "type": "modules/data-visualisation-app/SET_FILTERED_MATERIAL",
}
`;
